conda/gateways/repodata/__init__.py

```diff
@@ -534,14 +543,19 @@ def load(self, *, state_only=False) -> str:
             # it will release the lock early
             state = json.loads(state_file.read())
 
+            cache_path = self.cache_path_shards if binary else self.cache_path_json
+
             # json and state files should match. must read json before checking
             # stat (if json_data is to be trusted)
             if state_only:
-                json_data = ""
+                json_data = b"" if binary else ""
             else:
-                json_data = self.cache_path_json.read_text()
+                if binary:
+                    json_data = cache_path.read_bytes()
+                else:
+                    json_data = cache_path.read_text()
 
-            json_stat = self.cache_path_json.stat()
+            json_stat = cache_path.stat()
             if not (
                 state.get("mtime_ns") == json_stat.st_mtime_ns
```

Comment:
It might make sense to store a separate mtime, size, etag, etc. for the msgpack file, in the same state file, and ignore the other one when dealing with bytes.

Reply:
This code is designed to cache one of several remote files (repodata.json, repodata.json.zst in hte same file; repodata_shards.msgpack.zst in a different file) and compares the mtime to decide whether to send etag to the remote server.
Normally you are always fetching the same file once you decide which one is the best.
With one mtime that could refer to one of two local files we will definitely ignore our stored etag if we ever switch between formats, downloading a fresh copy of the file.